doc/howtos/bluetoothspeaker.rst

```diff
@@ -0,0 +1,141 @@
+Bluetooth Speaker
+#################
+
+How to setup your intel-iot-refkit image to function as a bluetooth speaker
+===========================================================================
+
+For configuring the bluetooth speaker setup you should use the
+"gateway" image and prepare couple of configuration files. This
+configuration has been tested to work with Intel Joule development
+board which has bluetooth radio chip integrated. The following later
+detailed 5 steps should do it:
+
+1. Start ``PulseAudio`` in the boot
+2. Add ``D-Bus`` configuration so that ``PulseAudio`` can communicate with ``BlueZ``
+3. Configure ``BlueZ`` to enable A2DP sink
+4. Unblock bluetooth at boot with ``ConnMan``
+5. Pair and connect your source device
+
+Start ``PulseAudio`` in the boot
+============================
+
+The way to start ``PulseAudio`` in the boot is to use systemd service
+file. In this example service file we start ``PulseAudio`` in system
+mode and don't let the daemon exit when in idle (which is normal
+behavior). ``PulseAudio`` will just go to "sleep"::
+
+  pulseaudio
+  [Unit]
+  Description=Pulse Audio
+
+  [Service]
+  Type=simple
+  ExecStart=/usr/bin/pulseaudio --system --disallow-exit --disable-shm --exit-idle-time=-1
+
+  [Install]
+  WantedBy=multi-user.target
+
+Copy the ``pulseaudio.service`` file to::
+
+  /etc/systemd/system/
+
+and enable it by::
+
+  systemctl enable pulseaudio.service
+
+For this particular use case we are starting ``PulseAudio`` in system
+mode and including bluetooth modules to the system.pa file. Remember
+that starting ``PulseAudio`` in system mode is not the recommended way
```

Comment:
Pulseaudio is usually started by the linux user. In this demo we don't have users for the bluetooth sw (except root), which makes it a little bit hard. Anyway pulseaudio is dropping its root priviledges and running as "pulse" so it should not be able to do too much harm to your speaker system. If somebody gets into a physical contact with the device game is anyway lost.